src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperDescriptorJsonConverter.cs

```diff
@@ -906,5 +914,141 @@ private static void ReadToEndOfCurrentObject(JsonReader reader)
 
             throw new JsonSerializationException($"Could not read till end of object, end of stream. Got '{reader.TokenType}'.");
         }
+
+        private static string Intern(string str)
+        {
+            if (str is null)
+            {
+                return null;
+            }
+
+            var interned = string.IsInterned(str);
```

Comment:
Will this ever return non-null? How will it know if we're only adding to our own dictionary?

Reply:
There are a bunch of C# strings which are being interned by other systems such as Fully-Qualified-Type-Names. I have confirmed that this path gets hit when launching a basic Razor project.